src/hex/roller.ts

```diff
@@ -0,0 +1,90 @@
+import * as Mustache from 'mustache';
+import {countMatches} from '../arrays';
+import {combineAll} from '../lang';
+import {RandomNumberGenerator} from '../rng';
+import {combineRolls, Roll, rollDie, Roller} from '../roller';
+import base from '../template';
+import {DieRollView} from '../view';
+import {
+    BLUT_ROLL_TABLE,
+    BONUS_ROLL_TABLE,
+    Dice,
+    DicePool,
+    dieRollImages,
+    ELIXIR_ROLL_TABLE,
+    Faces,
+    FLUCH_ROLL_TABLE,
+    HEXXEN_ROLL_TABLE,
+    interpretResult,
+    MALUS_ROLL_TABLE,
+    parseRollValues,
+    RollValues,
+    rollValuesMonoid,
+    SEGNUNG_ROLL_TABLE,
+} from './dice';
+import {SimpleParser} from './parser';
+import tpl from './template';
+
+export class HEXRoller extends Roller<Dice, Faces, DicePool> {
+
+    constructor(private rng: RandomNumberGenerator, command: string) {
+        super(command, [new SimpleParser()], true, true);
+    }
+
+    public roll(pool: DicePool): Array<Roll<Dice, Faces>> {
+        if (pool.BONUS >= pool.MALUS) {
+          pool.BONUS = pool.BONUS - pool.MALUS;
+          pool.MALUS = 0;
+        } else if (pool.MALUS > pool.BONUS) {
+          pool.MALUS = pool.MALUS - pool.BONUS;
+          pool.BONUS = 0;
+        }
+        return [
+            ...rollDie(pool.HEXXEN, Dice.HEXXEN, HEXXEN_ROLL_TABLE, this.rng),
+            ...rollDie(pool.BONUS, Dice.BONUS, BONUS_ROLL_TABLE, this.rng),
+            ...rollDie(pool.MALUS, Dice.MALUS, MALUS_ROLL_TABLE, this.rng),
+            ...rollDie(pool.SEGNUNG, Dice.SEGNUNG, SEGNUNG_ROLL_TABLE, this.rng),
+            ...rollDie(pool.BLUT, Dice.BLUT, BLUT_ROLL_TABLE, this.rng),
+            ...rollDie(pool.ELIXIR, Dice.ELIXIR, ELIXIR_ROLL_TABLE, this.rng),
+            ...rollDie(pool.FLUCH, Dice.FLUCH, FLUCH_ROLL_TABLE, this.rng),
+        ];
+    }
+
+    public combineRolls(rolls: Array<Roll<Dice, Faces>>): RollValues {
+        const results = rolls
+            .map((roll) => parseRollValues(roll));
+        return combineAll(results, rollValuesMonoid);
+    }
+
+    public toRoll(die: number, face: number): Roll<Dice, Faces> {
+        return new Roll(die, face);
+    }
+
+    public formatRolls(rolls: Array<Roll<Dice, Faces>>): string {
```

Comment:
This should also pass the flavor text like the other rollers do. I suppose you read the docs (which need to be updated for this change, forgot that)

Reply:
Changed that.